### PR TITLE
[MCHECKSTYLE-384] Update IT tests to support checkstyle 8.28

### PR DESCRIPTION
**Pull request by hazendaz**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MCHECKSTYLE) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MCHECKSTYLE-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MCHECKSTYLE-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/it/MCHECKSTYLE-357/pom.xml [hunk 1]
```diff
@@ -55,9 +60,7 @@
 
                                     <!-- Annotations -->
                                     <module name="AnnotationUseStyle" />
-                                    <module name="MissingDeprecated">
```
Comment (eolivelli) [inline src/it/MCHECKSTYLE-357/pom.xml]: are we losing this part ?

Comment (rnveach) [reply src/it/MCHECKSTYLE-357/pom.xml]: @eolivelli Please see https://checkstyle.org/releasenotes.html#Release_8.24
> MissingDeprecated: refactor to use javadoc AST 

and https://github.com/checkstyle/checkstyle/issues/4983 .

To me it looks the changes are just removing the property. property `skipNoJavadoc` was removed as the rewrite of the check will skip empty javadocs by default. If user wants to require javadocs we have a bunch of `MissingXxxJavadoc` checks.

**Comment by eolivelli** [comment]
I am not sure we should merge this change now that we are cutting 3.1.1
this release wants to open the way to using latest Checkstyle but without updating checkstyle itself

**Comment by hazendaz** [comment]
I'm not sure I entirely follow.  Everyone is  already told to override checkstyle for updates and adjust config if they override defaults.  This ensures the build is compliant with 8.28 and binary was set to 8.28.  That way anyone running into config issues see what was done here.  Do you mean the fact no new tests for any newer features? Are those required of this PR?  I think that would be a separate PR.

If 3.1.1 release is in flight this can come later.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by hazendaz** [comment]
I now believe the original concern is in-line support for new items.  That should be opened as a separate ticket.  The intent here is to only get the project building with 8.28 including integration tests which needed the handholding.  It appears to me looking at project that there are likely all kinds of items not supported directly given jump in 3.0.0 hardly did much with this library from 6.x checkstyle to 8.x checkstyle.  In fact, the 3.0.0 plugin release still supports with override up to checkstyle 8.24.

**Comment by romani** [comment]
@hazendaz , @eolivelli 
Security issue were found core library again and fix will be released as 8.29 at 26 Jan 2020.
Please plan to upgrade default version of checkstyle to 8.29.

**Comment by eolivelli** [comment]
@romano I understand.
It si a very food motivazione.
Se shuold chance the JIRA totale andrà the PR totale.

@hazendas what do you think?

**Comment by hazendaz** [comment]
@eolivelli I'd probably just merge this as-is for build purposes.  Then as 8.29 is out there, pull it separately.  Honestly, I sort of think in general this plugin should be getting released everytime checkstyles does so users can stop overriding the version included.  I don't think that goal would be too difficult and it would also ensure that new features are actually getting applied which may or may not be the case currently (haven't looked deep enough).

**Comment by romani** [comment]
8.29 is out

**Comment by eolivelli** [comment]
I am going to accept this PR and then bump checkstyle to 8.29.
I see github started complaining about checkstyle over every project that is using checkstyle maven plugin

**Comment by eolivelli** [comment]
Waiting for CI
https://builds.apache.org/job/maven-box/job/maven-checkstyle-plugin/job/MCHECKSTYLE-384/

**Comment by eolivelli** [comment]
I did not rebase and squash all of the commits.
I have re-launched CI

**Comment by eolivelli** [comment]
https://builds.apache.org/job/maven-box/job/maven-checkstyle-plugin/job/MCHECKSTYLE-384/

**Comment by eolivelli** [comment]
merged as e1e2f28620e2baed8351fea5fa524e98749df55c

thank you @hazendaz

**Pull request closed without merging**